### ISSUE TITLE
Fix for already connected error in ClientRound

### ISSUE THREAD

**Issue by andrew-aladev** [thaliproject/Thali_CordovaPlugin#1242]

**Comment by andrew-aladev** [comment]
Android doesn't like that we are trying to connect to already connected peer from previous test round.

**Comment by andrew-aladev** [comment]
https://gist.githubusercontent.com/vasilevskayaem/1b8cc8783be73082724a9f7f348e895a/raw/cf0187ae3e615c0a6900ad29731830c4eed088a8/gistfile1.txt

**Comment by andrew-aladev** [comment]
fixed

**Comment by yaronyg** [comment]
@vasilevskayaem hasn't had a chance to check this yet.

**Comment by yaronyg** [comment]
It looks like this is lines 215-218 of testThaliMobileNative in master, please investigate.

**Comment by yaronyg** [comment]
So my concern about failing on 'already connected' was bogus. The reason is that if we get an 'already connected' error in node this will cause connectToPeerPromise to fail but err.fatal in clientRound will e set to false in the catch and we will then reject again causing us to just delete the peerIdentifier from peersWeAreOrHaveResolved and thus allowing us to try again. This will cause us to go into a retry loop (exactly like we saw with Stanton) but that in and of itself shouldn't be harmful. We are just wasting cycles sending connect requests to Android that it will reject without ever bothering to even touch the Bluetooth stack since it will see that we are already connected to the peerID.

In other words 'already connected' is NOT a bug. It's a (cough) feature.

I also looked at the log btw and it doesn't actually show the test failing. All I can find in that log are 'ok'. Can you please show me a log where this is a problem and has a 'not ok'? Even if it's just a time out? Also please send me the coordinate logs and the logs for all 3 devices involved. I can't diagnose this without data!

**Comment by yaronyg** [comment]
We'll bring this back as part of the new round of testing that @evabishchevich is doing.
